server/odc-common/src/main/java/com/oceanbase/odc/common/util/Lazy.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (c) 2023 OceanBase.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.oceanbase.odc.common.util;
+
+import java.util.function.Supplier;
+
+import lombok.NonNull;
+
+/**
+ * @author: liuyizhuo.lyz
+ * @date: 2024/7/11
+ */
+public class Lazy<T> {
+    private static final Object NO_INIT = new Object();
```

Comment:
In my opinion, Lazy may not give accurate meaning of the class intend to do. This class want to provide a delayed function call if needed. It's more like a local cache, if missing then provide a value use provider.
If supplier call returns NO_INIT, get method will be always invoked. Get method seems not a high frequent access function. So move synchronized keyword to  method declare, use init flag to control supplier get seems more reasonable